lib/i18n/js/segment.rb

```diff
@@ -14,12 +16,21 @@ def initialize(file, translations, options = {})
 
       # Saves JSON file containing translations
       def save!
-        FileUtils.mkdir_p File.dirname(self.file)
+        if file =~ LOCALE_INTERPOLATOR
+          I18n.available_locales.each do |locale|
+            write_file(file_for_locale(locale), self.translations.slice(locale))
+          end
+        else
+          write_file(self.file, self.translations)
+        end
+      end
 
-        File.open(self.file, "w+") do |f|
+      def write_file(_file, _translations)
```

Comment:
Either make it a class method
Or make it `def write_file(_file = self.file, _translations = self.translations)` (I am not sure if this works)
It seems pointless to have an instance method 
that always takes some arguments that sometimes can be retrieved in the object itself